File: gems/test-miser/examples/zig/classifier_test.zig
const std = @import("std");
const classifier = @import("src/classifier.zig");

test "positive_primary" {
    try std.testing.expectEqualStrings("positive", classifier.classify(5));
}

test "positive_duplicate" {
    try std.testing.expectEqualStrings("positive", classifier.classify(5));
}

test "high" {
    try std.testing.expectEqualStrings("high", classifier.classify(11));
}

test "nonpositive" {
    try std.testing.expectEqualStrings("nonpositive", classifier.classify(0));
}

File: gems/test-miser/examples/go/classifier_test.go
package classifier

import "testing"

func TestPositivePrimary(t *testing.T) {
	if got := Classify(5); got != "positive" {
		t.Fatalf("got %q", got)
	}
}

func TestPositiveDuplicate(t *testing.T) {
	if got := Classify(5); got != "positive" {
		t.Fatalf("got %q", got)
	}
}

func TestHigh(t *testing.T) {
	if got := Classify(11); got != "high" {
		t.Fatalf("got %q", got)
	}
}

func TestNonpositive(t *testing.T) {
	if got := Classify(0); got != "nonpositive" {
		t.Fatalf("got %q", got)
	}
}

func TestSmoke(t *testing.T) {
	if true != true {
		t.Fatal("unreachable")
	}
}
